File: 0215-kth-largest-element-in-an-array/0215-kth-largest-element-in-an-array.py
class Solution:
    def findKthLargest(self, nums: List[int], k: int) -> int:
        
        hq = []
        for num in nums:
            heapq.heappush(hq, num)
            if len(hq) > k:
                heapq.heappop(hq)
        return hq[0]

File: 0215-kth-largest-element-in-an-array/0215-kth-largest-element-in-an-array.cpp
class Solution {
public:
    int findKthLargest(vector<int>& nums, int k) {
        std::priority_queue<int> pq;

        for(int i=0; i<nums.size(); i++) {
            pq.push(-nums[i]);
            if(pq.size() > k) {
                pq.pop();
            }
        }
        return -pq.top();
    }
};

/*

heap binary tree O(log n)

default max heap -> negative

while(k--)
    pop 


*/